roles/pulp-api/tasks/get_node_ip.yml

```diff
@@ -1,23 +1,53 @@
 ---
+
+- name: Set web protocol
+  set_fact:
+    web_protocol: "https"
+  when:
+    - ingress_tls_secret | length
+    - route_tls_secret | length
+
+- name: Set node address for ingress
+  set_fact:
+    node_address: "{{ web_protocol }}://{{ pulp_hostname }}"
+  when:
+    - ingress_type|lower == "ingress"
+
+- name: Set node address for route
+  set_fact:
+    node_address: "{{ web_protocol }}://{{ route_host }}"
+  when:
+    - ingress_type|lower == "route"
+
 - name: Look up the 1st address of the 1st node
   k8s_info:
     api_version: v1
     kind: Node
   register: k8s_nodes
+  when:
+    - node_address is not defined
 
 - name: DEBUG Show k8s_nodes
   debug:
     msg: "{{ k8s_nodes }}"
+  when:
+    - node_address is not defined
 
 - name: Set node address
   set_fact:
-    node_address: "{'content_origin': 'http://{{ k8s_nodes.resources[0].status.addresses[0].address }}:24816' }"
+    node_address: "http://{{ k8s_nodes.resources[0].status.addresses[0].address }}:24816"
+  when:
+    - node_address is not defined
+
+- name: Set content_origin
+  set_fact:
+    node_address: "{'content_origin': '{{ node_address }}' }"
```

Comment:
This is only used when ingress or route aren't selected.

Reply:
I can find out which plugin(s) need it set, but we (pulp-operator) still need to set it for those that plugin(s) though. Whether the user is using an ingress, a route, or none.

In fact, it needs to be set to the node that the webserver is running on if you are using none, but using a cluster. i.e., whatever users' miscellaneous yum/docker/ansible-galaxy clients talk to.

Reply:
so in the case of:
1)  **ingress** the `content_origin` is set to `pulp_hostname`
2) **route** the `content_origin` is set to `route_host`
3) **none** the `content_origin` is set to `node address`. I can set the port to 80 since pulp-web is a NodePort

I get that it was a hardcoded case to select the first node, but are we looking to really support pulp-operator in a case beyond development where they aren't using ingress or route?

Reply:
Practically all of them require it to be set.

https://docs.pulpproject.org/pulpcore/settings.html#content-origin

Reply:
I understand now.

Reply:
As for the none type, I think we should address that in a separate PR.

port 80 to the webserver would make sense, if it were not likely already being listened on.